libs/ledger-live-common/src/device/use-cases/appDataBackup/types.ts

```diff
@@ -0,0 +1,59 @@
+import { type AppStorageInfo, appStorageInfoGuard } from "@ledgerhq/device-core";
+import { DeviceModelId } from "@ledgerhq/devices";
+import { createCustomErrorClass } from "@ledgerhq/errors";
+
+export interface StorageProvider<VType> {
+  getItem(key: string): Promise<VType | null>;
+  setItem(key: string, value: VType): Promise<void>;
+  removeItem(key: string): Promise<void>;
+}
+
+export type AppName = string;
+export type AppStorageKey =
+  `${Exclude<DeviceModelId, DeviceModelId.blue | DeviceModelId.nanoS>}-${AppName}`;
+
+export type AppStorageType = {
+  appDataInfo: AppStorageInfo;
+  appData: string;
+};
+
+export function appStorageTypeGuard(data: AppStorageType | unknown): data is AppStorageType {
+  return (
+    typeof data === "object" &&
+    data !== null &&
+    "appDataInfo" in data &&
+    appStorageInfoGuard(data.appDataInfo) &&
+    "appData" in data &&
+    typeof data.appData === "string"
+  );
+}
+
+export enum BackupAppDataEventType {
+  Progress = "progress",
+  AppDataInfoFetched = "appDataInfoFetched",
+  AppDataBackedUp = "appDataBackedUp",
+  AppDataAlreadyBackedUp = "appDataAlreadyBackedUp",
+  NoAppDataToBackup = "noAppDataToBackup",
+}
+
+export type BackupAppDataEvent =
+  | {
+      type: BackupAppDataEventType.Progress;
+      data: number; // Progress percentage
```

Comment:
[SHOULD] Use TSDoc for those comments.

Reply:
Also I would rather name this key `progress`, and the one below for `AppDataInfoFetched`, `appStorageInfo`.

Reply:
This aimes to have the same format for event `{ type: Enum, data: DataType }` and it's easy to maintain and treat an event without knowing the specific property name of each event type. When someone wants to treat an event of type progress, he/she must know the event type is `BackupAppDataEventType.Progress`, so the data cannot be something else. In terms of security, the event type defines strictly the data type.

Reply:
Yep yep it's just for readability but what you're saying makes sense too, as you want 👍